### PR TITLE
[risk=low] RW-10622 refactor proxyurl

### PR DESCRIPTION
**Pull request by Qi77Qi**
https://precisionmedicineinitiative.atlassian.net/browse/RW-10622

new getApp response will look like this
```
{
  "appName": "qi-rstudio-812",
  "cloudContext": {
    "cloudProvider": "GCP",
    "cloudResource": "terra-quality-bf8ac05e"
  },
  "kubernetesRuntimeConfig": {
    "numNodes": 1,
    "machineType": "n1-highmem-8",
    "autoscalingEnabled": false
  },
  "errors": [],
  "status": "RUNNING",
  "proxyUrls": {
    "app": "https://leonardo.qi-aou2.bee.envs-terra.bio/proxy/google/v1/apps/terra-quality-bf8ac05e/qi-rstudio-812/app",
    "welder-service": "https://leonardo.qi-aou2.bee.envs-terra.bio/proxy/google/v1/apps/terra-quality-bf8ac05e/qi-rstudio-812/welder-service"
  },
  "diskName": "qi-rstudio-812-disk",
  "customEnvironmentVariables": {
    "WORKSPACE_NAME": "qi-ws"
  },
  "auditInfo": {
    "creator": "qiwangterra@gmail.com",
    "createdDate": "2023-08-01T17:52:21.890069Z",
    "destroyedDate": null,
    "dateAccessed": "2023-08-01T17:52:21.890069Z"
  },
  "appType": "ALLOWED",
  "chartName": "/leonardo/aou-rstudio-chart",
  "accessScope": null,
  "labels": {
    "clusterServiceAccount": "aic2pet-2689964849996203093b6@terra-quality-bf8ac05e.iam.gserviceaccount.com",
    "appName": "qi-rstudio-812",
    "creator": "qiwangterra@gmail.com",
    "googleProject": "terra-quality-bf8ac05e"
  }
}
```

Tested locally

https://github.com/all-of-us/workbench/assets/32771737/f67f46db-f5eb-4973-b5cc-f44ea48fdbf7

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: ui/src/app/components/apps-panel/expanded-app.spec.tsx [hunk 1]
```diff
@@ -355,7 +355,7 @@ describe('ExpandedApp', () => {
       appType: AppType.RSTUDIO,
       status: AppStatus.RUNNING,
       proxyUrls: {
-        'rstudio-service': proxyUrl,
```
Comment (jmthibault79) [inline ui/src/app/components/apps-panel/expanded-app.spec.tsx:358]: would be nice to have a constant for this

Comment (Qi77Qi) [reply ui/src/app/components/apps-panel/expanded-app.spec.tsx:358]: I'm down for that..do u have any suggestion as to where I should put it? any existing examples?

Comment (jmthibault79) [reply ui/src/app/components/apps-panel/expanded-app.spec.tsx:358]: there's a `constants` file but I'm not sure if that's appropriate... maybe someplace like an analysis-utils file (or analysis/utils?)

File: ui/src/app/components/apps-panel/expanded-app.spec.tsx [hunk 2]
```diff
@@ -42,6 +42,7 @@ import { ALL_GKE_APP_STATUSES, minus } from 'testing/utils';
 
 import { ExpandedApp } from './expanded-app';
 import { UIAppType } from './utils';
+import {PROXY_PATH_SUFFIX} from "../../utils/constants";
```
Comment (jmthibault79) [inline ui/src/app/components/apps-panel/expanded-app.spec.tsx:45]: the linter's not going to happy about this: run `eslint` in IntelliJ or `yarn lint --fix` at the CLI

File: ui/src/app/utils/constants.tsx [hunk 3]
```diff
@@ -87,3 +87,6 @@ export const STATE_CODE_MAPPING = {
 export const JUPYTER_FILE_EXT = '.ipynb';
 export const RMD_FILE_EXT = '.Rmd';
 export const R_SCRIPT_EXT = '.R';
+export const R_FILE_EXT = '.R';
+
+export const PROXY_PATH_SUFFIX = 'app';
```
Comment (jmthibault79) [inline ui/src/app/utils/constants.tsx:92]: This change doesn't affect the Jupyter runtime proxy path, right?  We should name this accordingly, with GKE_ or similar

File: ui/src/testing/stubs/apps-api-stub.ts [hunk 4]
```diff
@@ -46,7 +46,7 @@ const createRStudioListAppsResponseDefaults: UserAppEnvironment = {
   appName: 'all-of-us-2-rstudio-1234',
   appType: AppType.RSTUDIO,
   proxyUrls: {
-    'rstudio-service':
+    PROXY_PATH_SUFFIX:
       'https://leonardo.dsde-dev.broadinstitute.org/proxy/google/v1/apps/terra-vpc-sc-dev-1234/all-of-us-2-rstudio-1234/rstudio',
```
Comment (jmthibault79) [inline ui/src/testing/stubs/apps-api-stub.ts:50]: let's update the path returned as well

File: ui/src/app/utils/constants.tsx [hunk 5]
```diff
@@ -87,3 +87,6 @@ export const STATE_CODE_MAPPING = {
 export const JUPYTER_FILE_EXT = '.ipynb';
 export const RMD_FILE_EXT = '.Rmd';
 export const R_SCRIPT_EXT = '.R';
+export const R_FILE_EXT = '.R';
```
Comment (jmthibault79) [inline ui/src/app/utils/constants.tsx:90]: this line is a rebasing accident - can remove it

**Review by jmthibault79 — COMMENTED** [review]
Looks good.

But we can't test it yet because the corresponding change is not yet live on Leonardo.  I'd prefer to wait to merge until then.

**Comment by Qi77Qi** [comment]
Yea. I'll test things out first before merging

File: api/src/main/java/org/pmiops/workbench/utils/mappers/LeonardoMapper.java [hunk 6]
```diff
@@ -246,6 +246,7 @@ LeonardoKubernetesRuntimeConfig toLeonardoKubernetesRuntimeConfig(
 
   @ValueMapping(source = "GALAXY", target = MappingConstants.NULL) // we don't support Galaxy
   @ValueMapping(source = "CUSTOM", target = MappingConstants.NULL) // we don't support CUSTOM apps
+  @ValueMapping(source = "ALLOWED", target = "RSTUDIO") // we don't support CUSTOM apps
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/utils/mappers/LeonardoMapper.java:249]: comment is incorrect

**Review by jmthibault79 — APPROVED** [review]
Tested locally.  I can run both Cromwell and RStudio apps!

**Pull request merged**